ipa-macros/src/derive_step/mod.rs

```diff
@@ -112,20 +114,30 @@ pub fn expand(input: TokenStream) -> TokenStream {
 fn impl_as_ref(ident: &syn::Ident, data: &syn::DataEnum) -> Result<TokenStream2, syn::Error> {
     let mut const_arrays = Vec::new();
     let mut arms = Vec::new();
+    let mut res = Ok(());
 
     data.variants.iter().for_each(|v| {
         let ident = &v.ident;
         let ident_snake_case = ident.to_string().to_snake_case();
         let ident_upper_case = ident_snake_case.to_uppercase();
 
         if is_dynamic_step(v) {
-            // create an array of 64 strings and use the variant index as array index
-            let steps = (0..64)
+            let num_steps = match get_dynamic_step_count(v) {
+                Ok(n) => n,
+                Err(e) => {
+                    // we can't return from a closure, so we need to set the result and break
```

Comment:
I'd probably convert to `for` loop here to avoid this quirkiness

Reply:
done!